core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/execution/OperatorExecution.scala

```diff
@@ -44,33 +45,32 @@ case class OperatorExecution() {
   def getWorkerIds: Set[ActorVirtualIdentity] = workerExecutions.keys.asScala.toSet
 
   def getState: WorkflowAggregatedState = {
-    val workerStates = workerExecutions.values.asScala.map(_.getState).toArray
-    if (workerStates.isEmpty) {
-      return WorkflowAggregatedState.UNINITIALIZED
-    }
-    if (workerStates.forall(_ == COMPLETED)) {
-      return WorkflowAggregatedState.COMPLETED
-    }
-    if (workerStates.contains(RUNNING)) {
-      return WorkflowAggregatedState.RUNNING
-    }
-    val unCompletedWorkerStates = workerStates.filter(_ != COMPLETED)
-    if (unCompletedWorkerStates.forall(_ == UNINITIALIZED)) {
-      WorkflowAggregatedState.UNINITIALIZED
-    } else if (unCompletedWorkerStates.forall(_ == PAUSED)) {
-      WorkflowAggregatedState.PAUSED
-    } else if (unCompletedWorkerStates.forall(_ == READY)) {
-      WorkflowAggregatedState.READY
-    } else {
-      WorkflowAggregatedState.UNKNOWN
-    }
+    val workerStates = workerExecutions.values.asScala.map(_.getState.value)
+    aggregateStates(
+      workerStates,
+      WorkerState.COMPLETED.value,
+      WorkerState.RUNNING.value,
+      WorkerState.UNINITIALIZED.value,
+      WorkerState.PAUSED.value,
+      WorkerState.READY.value
+    )
   }
 
   def getStats: OperatorRuntimeStats =
     OperatorRuntimeStats(
       getState,
-      inputCount = workerExecutions.values.asScala.map(_.getStats).map(_.inputTupleCount).sum,
-      outputCount = workerExecutions.values.asScala.map(_.getStats).map(_.outputTupleCount).sum,
+      inputCount = workerExecutions.values.asScala
```

Comment:
is this logic the same with `ExecutionUtils.aggregateStats`? I see one difference one is using `OperatorRuntimeStats.inputCount` and the other one is using `WorkerInfo` (`WorkerMetrics.getStats.inputTupleCount`). if the logic is the same, is it easy to merge them?

Reply:
It looks difficult to merge them because WorkerMetrics and OperatorMetrics are different.